collector_test.go

```diff
@@ -2,26 +2,180 @@ package main
+		oidToPdu        map[string]gosnmp.SnmpPDU
+		expectedMetrics map[string]string
 	}{
 		{
 			pdu: &gosnmp.SnmpPDU{
 				Name:  "1.1.1.1.1",
+				Value: "SomeStringValue",
+			},
+			indexOids: []int{},
+			metric: &config.Metric{
+				Name: "TestMetricName",
+				Oid:  "1.1.1.1.1",
+				Help: "HelpText",
+				RegexpExtracts: map[string][]config.RegexpExtract{
+					"Extension": []config.RegexpExtract{
+						{
+							Regex: config.Regexp{
+								regexp.MustCompile(".*"),
+							},
+							Value: "5",
+						},
+					},
+				},
+			},
+			oidToPdu: make(map[string]gosnmp.SnmpPDU),
+			expectedMetrics: map[string]string{
+				`gauge:<value:5 > `: `Desc{fqName: "TestMetricNameExtension", help: "HelpText (regex extracted)", constLabels: {}, variableLabels: []}`,
+			},
+		},
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.1.1.1.1",
+				Value: "SomeStringValue",
+			},
+			indexOids: []int{},
+			metric: &config.Metric{
+				Name: "TestMetricName",
+				Oid:  "1.1.1.1.1",
+				Help: "HelpText",
+				RegexpExtracts: map[string][]config.RegexpExtract{
+					"Extension": []config.RegexpExtract{
+						{
+							Regex: config.Regexp{
+								regexp.MustCompile(".*"),
+							},
+							Value: "",
+						},
+					},
+				},
+			},
+			expectedMetrics: map[string]string{},
+		},
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.1.1.1.1",
+				Value: "SomeStringValue",
+			},
+			indexOids: []int{},
+			metric: &config.Metric{
+				Name: "TestMetricName",
+				Oid:  "1.1.1.1.1",
+				Help: "HelpText",
+				RegexpExtracts: map[string][]config.RegexpExtract{
+					"Extension": []config.RegexpExtract{
+						{
+							Regex: config.Regexp{
+								regexp.MustCompile("(will_not_match)"),
+							},
+							Value: "",
+						},
+					},
+				},
+			},
+			expectedMetrics: map[string]string{},
+		},
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.1.1.1.1",
+				Value: 2,
+			},
+			indexOids: []int{},
+			metric: &config.Metric{
+				Name: "TestMetricName",
+				Oid:  "1.1.1.1.1",
+				Help: "HelpText",
+				RegexpExtracts: map[string][]config.RegexpExtract{
+					"Status": []config.RegexpExtract{
+						{
+							Regex: config.Regexp{
+								regexp.MustCompile(".*"),
+							},
+							Value: "5",
+						},
+					},
+				},
+			},
+			expectedMetrics: map[string]string{},
+		},
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.1.1.1.1",
+				Value: "Test value 4.42 123 999",
+			},
+			indexOids: []int{},
+			metric: &config.Metric{
+				Name: "TestMetricName",
+				Oid:  "1.1.1.1.1",
+				Help: "HelpText",
+				RegexpExtracts: map[string][]config.RegexpExtract{
+					"Blank": []config.RegexpExtract{
+						{
+							Regex: config.Regexp{
+								regexp.MustCompile("XXXX"),
+							},
+							Value: "4",
+						},
+					},
+					"Extension": []config.RegexpExtract{
+						{
+							Regex: config.Regexp{
+								regexp.MustCompile(".*"),
+							},
+							Value: "5",
+						},
+					},
+					"MultipleRegexes": []config.RegexpExtract{
```

Comment:
I'd have three regexes, and make the second one match - but the 3rd one would also match. That way we know it's trying them in turn.